### PR TITLE
WIP: CNY theme + theme switcher

### PR DESCRIPTION
**Pull request by svub**
First version of CNY + holiday cards with theme switcher. 

There are issues with the background (the SVG is 8 MB...), so the background in this PR is a quick hack to have something to test with.

Red envelopes will be handed out on Jan 24 during CNY, so we should start testing this online early next week to have it ready mid of the week latest. Help appreciated. :)

### CODE REVIEW

**Comment by sisou** [comment]
The big background image should be possible to separate into a background-pattern, that can be applied with CSS to the document. The lamps could be separated into the same shape, and re-used three times, with different sizes and opacities.

Futhermore, the original SVG should be put through https://jakearchibald.github.io/svgomg/ to minify.

**Comment by sisou** [comment]
The card background can probably also be optimized by re-using a gradient for the letters and greatly simpifying it's path, instead of having 20 gradients with complex paths defined.

**Comment by svub** [comment]
Yep, Kwodwo is on it. Quick question @sisou: do you know if it's possible to have a repeating pattern within the SVG so that we don't need special code to handle this? Would like to keep everything generic and just change the images of possible.

Side note: I was planning to extract the `Dropdown` into the vue-components repo for reuse. Made it generic. Let me know what you think and if just putting it in a branch and send a wip PR is the way to get this started.

**Comment by sisou** [comment]
Yes, SVGs can contain repeating patterns in itself, with the fittingly named `<pattern>` element:
https://developer.mozilla.org/en-US/docs/Web/SVG/Tutorial/Patterns

For the vue-component component, just create a PR in that repo, yes :+1:

**Comment by danimoh** [comment]
About the svg:
It's the background with the waves that contributes most to the size of the svg.
By just minifying and removing the gradients I managed to get this down to 979kb (87kb gzipped) without really noticeable difference.
Using a pattern would still definitely be preferable and lead to a much smaller size but if you don't manage to do so, the gzipped size might already be acceptable.

Optimizing the lamps can be done (I guess Soeren was hinting at [`<use>` Elements](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use)) but has less effect on the file size.

Note also that optimizing the card image is off less importance than the background as it's only displayed in the second step and thus has more time to be prefetched in the background.

File: src/App.vue [hunk 1]
```diff
@@ -1,37 +1,46 @@
 <template>
     <main id="app">
+        <!-- TODO remove? update to themes/ folder? -->
         <link rel="prefetch" href="img/christmas-card.svg"><!-- this is the path in the deployment -->
```
[the remark this answers is not in the document]
Comment (svub) [reply src/App.vue]: Nice! Just didn't know how to make it work again. Solution via `.env` file is perfect!

File: src/Dropdown.vue [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                <path d="M3 .7c.3-.3.7-.3 1 0l2.6 4.5c.2.4 0 .8-.5.8H1a.5.5 0 01-.5-.8L3.1.7z"/>
+            </svg>
+        </button>
+        <transition name="dropdown-fade">
+            <div class="background" v-if="extended">
+                <span
+                    v-for="other in others"
+                    :key="other.value"
+                    class="nq-label other"
+                    @click="select(other.value); extended = false"
```
[the remark this answers is not in the document]
Comment (svub) [reply src/Dropdown.vue]: Thought the same at first, but if I want to programmatically select something, I might not always want to close the dropdown. No strong opinion, but thought it's good to keep them separated as they are not necessarily the same.

File: src/Dropdown.vue [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+type ValueLabel = {
+    value: string,
+    label: string,
+}
+
+@Component({ components: {} })
+export default class Dropdown extends Vue {
+    @Prop({ type: Object, required: true }) private values?: { [value: string]: string };
+    @Prop(String) private default?: string;
+    @Prop({ type: String, default: 'green' }) private color?: string;
```
[the remark this answers is not in the document]
Comment (svub) [reply src/Dropdown.vue]: public is default, right?

File: src/Dropdown.vue [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                </span>
+            </div>
+        </transition>
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator';
+
+type ValueLabel = {
```
[the remark this answers is not in the document]
Comment (svub) [reply src/Dropdown.vue]: I was thinking and thinking, yep 'entries' is good!

File: src/Dropdown.vue [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    }
+
+    get selectedLabel() {
+        return (this.valueLabels.find(pair => pair.value === this.selected) || { label: 'empty' }).label;
+    }
+
+    get valueLabels(): ValueLabel[] {
+        if (this.values) {
+            if (Array.isArray(this.values)) {
+                return this.values.map(value => ({ value: value.replace(/\W/, ''), label: value }));
```
[the remark this answers is not in the document]
Comment (svub) [reply src/Dropdown.vue]: Any label would be a valid value/ID?

**Comment by svub** [comment]
Thanks @danimoh, great review! All fixes in App pushed, some in Dropdown done. Things is now, can we move your comments to the a PR to vue-components? I don't want to lose them but don't have the time to fix all of them for the CNY release. Do you know if that's possible and how to do it?

**Comment by svub** [comment]
About the background: new SVG uses `pattern` and is thus only 87kb *unzipped* :)

**Pull request merged**